tutorials/migration.md

```diff
@@ -4,96 +4,846 @@ layout: intro
+openHAB 2. Make sure you include these in your installation as well.
+
+###Installing Unofficially Supported openHAB 1.x Add-ons
+
+Skip this section if all the add-ons you need have been installed already.
+
+First check the [list of add-ons that are known not to work in openHAB 2](../addons/1xaddons.html#currently-incompatible-1x-add-ons)
+and make sure yours is not among them.
+
+Next install the openHAB 1.x compatibility layer using the Karaf Console 
+instructions above.
+
+Copy your openhab.cfg file to `<openHAB 2 conf>/services`. If you are running an apt-get 
+installed openHAB 1.x openhab.cfg is located in `/etc/openhab/configurations`.
+For an apt-get installed openHAB 2 the `<openHAB 2 conf>` folder is located in 
+`/etc/openhab2`. For manually installed openHAB 1.x this file is located
+in `<openhab home>/configurations` and in manually installed openHAB 2 it is
+`<openhab home>/conf`.
+
+This file will have a lot of redundant configuration information in it that you
+have already moved over to individual binding .cfgs. Make sure to comment those
+out and just leave the config parameters for the bindings you are manually 
+installing.
+
+Now copy the add-on's jar files that you want to install to the `<openHAB conf>/addons` folder for 
+openHAB 2. For example:
+
+```
+cp /usr/share/openhab/addons/org.openhab.binding.astro-1.8.3.jar /usr/share/openhab2/addons
+```
+
+There may be some errors about there not being a `bind` and `unbind` method. If
+everything worked you should see in the log the same entries from the binding 
+you see when openHAB 1.x starts.
+
+Watch the logs for errors as they can be informative. For example, some Actions
+require a corresponding Binding be installed first. An example error of this 
+sort looks like:
+
+```
+2016-09-08 15:15:04.613 [WARN ] [org.apache.felix.fileinstall        ] - Error while starting bundle: file:/openhab/addons/org.openhab.action.astro-1.8.3.jar
+org.osgi.framework.BundleException: Could not resolve module: org.openhab.action.astro [210]
+  Unresolved requirement: Import-Package: org.openhab.binding.astro.internal.calc
+```
+
+Note, the above error is for illustration purposes. You should not be 
+installing Astro this way.
+
+###Final Add-ons Installation Steps
+
+openHAB 2 has a different 
+[folder layout](http://docs.openhab.org/installation/linux.html#file-locations).
+Of particular node are the configuration folders and userdata folders. See the
+previous link for the location of these folders for your installtion.
+
+As with openHAB 1.x, one must restart openHAB 2 to pick up changes to .cfg files.
+Therefore restart openHAB 2 now.
+
+If you haven't already, configure default persistence now. You can do this 
+through `runtime.cfg` or PaperUI.
+
+Now openHAB 2 is ready to receive your other configurations, items and such. 
+Watch the log file as you transfer over files for errors. These errors will 
+have to be fixed.
+
+Copy over your files in the following order:
+
+~~~~
```

Comment:
It didn't work. The only way I could get it into code formatting was to use ~~~~